scripts/west_commands/sbom/README.rst

```diff
@@ -0,0 +1,361 @@
+already has licenses detected by some other previously executed detector.
+Detectors are executed from left to right using a list provided by the ``--license-detectors``.
+
+  .. code-block:: bash
+
+     --optional-license-detectors scancode-toolkit
+
+Some detectors may run in parallel on all available CPU cores, which speeds up the detection time.
+Use the ``-n`` option to limit the number of parallel threads or processes.
+
+.. _west_sbom HTML report overview:
+
+HTML report overview
+********************
+
+The HTML report has following structure:
+
+* Summary of the report. It contains:
+
+   * At the beginning are notes.
+     They will give you some general information that you should pay attention while reading the report.
+   * The list of inputs.
+     It shows from where this report takes the files.
+   * The list of licenses.
+     It contains all licenses detected in the input files.
+   * The list of added license texts.
+     If a license is not in the `SPDX License List`_ and we have it in our internal database,
+     the license text will be added to the report.
+
+  You can click links in the summary to get more details about specific item.
+
+* List of files without any license information or with license information that cannot be detected automatically.
+  You have to investigate them manually to get the license information.
+
+* Details about each detected license:
+
+   * License identifier
+   * Information if it is a standard SPDX license
+   * License name if available
+   * Link to license text or more details if available
+   * All files from the input covered by this license
+
+* License texts added to this report.
+
+.. _west_sbom Extracting from build:
+
+Extracting list of files from a build directory
+***********************************************
+
+The ``ncs-sbom`` extracts a list of files from a build directory by querying the ``ninja`` about its targets and dependencies.
+
+The entry point is the ``zephyr/zephyr.elf`` target.
+The script asks ``ninja`` for all input targets of the ``zephyr/zephyr.elf`` target.
+Next, it asks for all input targets of the previously extracted input targets,
+and so on until it reaches all leaves in the dependency tree.
+The result is a list of all the leaves.
+
+You can change the target or specify multiple targets by adding them after the build directory in the ``-d`` option, for example:
+
+.. parsed-literal::
+   :class: highlight
+
+   -d build_directory *target1.elf* *target2.elf*
+
+Two redundant methods for increasing the correctness of the above algorithm are implemented:
+
+* Each library is examined using the GNU ``ar`` tool.
+  If the list of files returned by the GNU ``ar`` tool is covered by the list returned from the ``ninja``,
+  the list is assumed to be valid.
+  Otherwise, the library is assumed to be a leaf, so it is shown in the report and its inputs are not analyzed further.
```

Comment:
```suggestion
* Each library is examined using the GNU ``ar`` tool.

  If the list of files returned by the GNU ``ar`` tool is covered by the list returned from the ``ninja``, the list is assumed to be valid.
  Otherwise, the library is assumed to be a leaf, so it is shown in the report and its inputs are not analyzed further.
```